scipy/fft/tests/test_fftlog.py

```diff
@@ -0,0 +1,154 @@
+               -0.7834031308271878E-03,  0.3931444945110708E-03,
+               -0.2697710625194777E-03,  0.3568398050238820E-03,
+               -0.5554454827797206E-03,  0.8286331026468585E-03 ]
+    assert_allclose(ours, theirs)
+
+    # test 2: change to optimal offset
+    offset = fhtoffset(dln, mu, bias=bias)
+    ours = fht(a, dln, mu, offset=offset, bias=bias)
+    theirs = [  0.4353768523152057E-04, -0.9197045663594285E-05,
+                0.3150140927838524E-03,  0.9149121960963704E-03,
+                0.5808089753959363E-02,  0.2548065256377240E-01,
+                0.1339477692089897    ,  0.4821530509479356    ,
+                0.2659899781579785    , -0.1116475278448113E-01,
+                0.1791441617592385E-02, -0.4181810476548056E-03,
+                0.1314963536765343E-03, -0.5422057743066297E-04,
+                0.3208681804170443E-04, -0.2696849476008234E-04 ]
+    assert_allclose(ours, theirs)
+
+    # test 3: positive bias
+    bias = 0.8
+    offset = fhtoffset(dln, mu, bias=bias)
+    ours = fht(a, dln, mu, offset=offset, bias=bias)
+    theirs = [ -7.343667355831685     ,  0.1710271207817100    ,
+                0.1065374386206564    , -0.5121739602708132E-01,
+                0.2636649319269470E-01,  0.1697209218849693E-01,
+                0.1250215614723183    ,  0.4739583261486729    ,
+                0.2841149874912028    , -0.8312764741645729E-02,
+                0.1024233505508988E-02, -0.1644902767389120E-03,
+                0.3305775476926270E-04, -0.7786993194882709E-05,
+                0.1962258449520547E-05, -0.8977895734909250E-06 ]
+    assert_allclose(ours, theirs)
+
+    # test 4: negative bias
+    bias = -0.8
+    offset = fhtoffset(dln, mu, bias=bias)
+    ours = fht(a, dln, mu, offset=offset, bias=bias)
+    theirs = [  0.8985777068568745E-05,  0.4074898209936099E-04,
+                0.2123969254700955E-03,  0.1009558244834628E-02,
+                0.5131386375222176E-02,  0.2461678673516286E-01,
+                0.1235812845384476    ,  0.4719570096404403    ,
+                0.2893487490631317    , -0.1686570611318716E-01,
+                0.2231398155172505E-01, -0.1480742256379873E-01,
+                0.1692387813500801    ,  0.3097490354365797    ,
+                2.759360718240186     , 10.52510750700458       ]
+    assert_allclose(ours, theirs)
+
+
+@pytest.mark.parametrize('optimal', [True, False])
+@pytest.mark.parametrize('offset', [0.0, 1.0, -1.0])
+@pytest.mark.parametrize('bias', [0, 0.1, -0.1])
+@pytest.mark.parametrize('n', [64, 63])
+def test_fht_identity(n, bias, offset, optimal):
+    a = np.random.randn(n)
+    dln = np.random.uniform(-1, 1)
+    mu = np.random.uniform(-2, 2)
+
+    if optimal:
+        offset = fhtoffset(dln, mu, initial=offset, bias=bias)
+
+    A = fht(a, dln, mu, offset=offset, bias=bias)
+    a_ = ifht(A, dln, mu, offset=offset, bias=bias)
+
+    assert_allclose(a, a_)
+
+
+def test_fht_special_cases():
+    a = np.random.randn(64)
```

Comment:
I wonder if it is advisable to pin down the random seed here and in other new tests that use `random` infrastructure?

I also noticed that recent numpy docs suggest a preferred alternative to `randn` (Robert et al. probably have more to say about than than I would though)
![image](https://user-images.githubusercontent.com/7903078/118380663-b3bf8500-b5a0-11eb-8bc3-5caf7d04d410.png)

Reply:
Yes, set a seed at the top of the function with `np.random.seed(3491349965)` (just so we don't have to track down weird intermittent failures when the PRNG throws you some odd edge case), and use `np.random.standard_normal()` instead of `randn()`. Tests can continue to use `np.random.seed()` and the `np.random` convenience functions to generate test data.

If you want to write a little more modern code, use `rng = np.random.RandomState(3491349965)` and call `rng.standard_normal()`, etc. methods instead of the `np.random` convenience functions.

With the way that you are using the data here, you could probably safely use `rng = np.random.default_rng(3491349965)`, but I'm not going to press it.

Reply:
If this is the new way to set seed I'm quite happy to start using it actually. Previously it was buried down under the discussion hence sorry for my noise.

Reply:
I've implemented the first proposed change. You could also introduce a `rng` fixture that does all of this in a sensible automatic manner with zero overhead for the test writer.